mysql/datadog_checks/mysql/mysql.py

```diff
@@ -1213,3 +1230,22 @@ def _report_warnings(self):
 
         for warning in messages:
             self.warning(warning)
+
+    def _send_database_instance_metadata(self):
+        if self.resolved_hostname not in self._database_instance_emitted:
+            event = {
+                "host": self.resolved_hostname,
+                "agent_version": datadog_agent.get_version(),
+                "dbms": "mysql",
+                "kind": "database_instance",
+                "collection_interval": self._config.database_instance_collection_interval,
+                'dbms_version': self.version.version + '+' + self.version.build,
+                "tags": self._non_internal_tags,
+                "timestamp": time.time() * 1000,
+                "cloud_metadata": self._config.cloud_metadata,
+                "metadata": {
+                    "dbm": self._config.dbm_enabled,
+                    "connection_host": self._config.host,
+                },
+            }
+            self.database_monitoring_metadata(json.dumps(event, default=default_json_event_encoding))
```

Comment:
Add the `event` to the cache before emit?
```
self. _database_instance_emitted[self.resolved_hostname] = event
```

Reply:
Ah yeah, good catch. Thanks `:high-five:`.